src/android/interoperability/cpp/cpp-bridge.md

```diff
@@ -0,0 +1,52 @@
+# C++ Bridge Declarations
+
+```rust,ignore
+#[cxx::bridge]
+mod ffi {
+{{#include ../../../../third_party/cxx/blobstore/src/main.rs:cpp_bridge}}
+}
+```
+
+Results in (roughly) the following Rust:
+
+```rust,ignore
+#[repr(C)]
+pub struct BlobstoreClient {
+    _private: ::cxx::private::Opaque,
+}
+
+pub fn new_blobstore_client() -> ::cxx::UniquePtr<BlobstoreClient> {
+    extern "C" {
+        #[link_name = "org$blobstore$cxxbridge1$new_blobstore_client"]
+        fn __new_blobstore_client() -> *mut BlobstoreClient;
+    }
+    unsafe { ::cxx::UniquePtr::from_raw(__new_blobstore_client()) }
+}
+
+impl BlobstoreClient {
+    pub fn put(&self, parts: &mut MultiBuf) -> u64 {
+        extern "C" {
+            #[link_name = "org$blobstore$cxxbridge1$BlobstoreClient$put"]
+            fn __put(
+                _: &BlobstoreClient,
+                parts: *mut ::cxx::core::ffi::c_void,
+            ) -> u64;
+        }
+        unsafe {
+            __put(self, parts as *mut MultiBuf as *mut ::cxx::core::ffi::c_void)
+        }
+    }
+}
+
+// ...
+```
+
+<details>
+
+* The programmer does not need to promise that the signatures they have typed in
+  are accurate; that would be unreasonable. CXX performs static assertions that
```

Comment:
More specifically it seems like it would make API changes down the road require remembering to manually recheck signatures, which could be a significant source of bugs.

Reply:
I'm not sure what you're referring to here?

Reply:
I mean that the justification is not just that it would be "unreasonable" but more specifically that it would be a source of bugs to trust the programmer's signatures unconditionally.

Reply:
Ah sure. Removing the weird wording.